### PR TITLE
feat(gen9): Wave 2 -- Terastallization BattleGimmick + calculateTeraStab

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary

- Implement `Gen9Terastallization` class (`BattleGimmick` interface): `canUse()`, `activate()`, `modifyMove()` (Tera Blast type/category/BP changes)
- Export `calculateTeraStab()` helper with full STAB logic: standard Tera (2.0x match, 1.5x partial), Stellar (one-time 2x per base type, 4915/4096 non-base), Adaptability interaction (1.5x->2.0x, 2.0x->2.25x)
- Add Tera persistence through switches: new `PokemonInstance` fields (`terastallized`, `teraTypes`, `stellarBoostedTypes`) + `createActivePokemon()` restoration logic
- 44 new tests (100 total in gen9) covering canUse gates, activate mutations, Tera Blast modifyMove, all STAB scenarios, Stellar tracking, Adaptability edge cases, and switch persistence

## Test plan

- [x] canUse() returns false when gimmickUsed, already terastallized, or no teraType
- [x] activate() sets correct types, persistence fields, emits TerastallizeEvent
- [x] Stellar Tera retains original types defensively
- [x] modifyMove() transforms Tera Blast (type, category, Stellar 100 BP)
- [x] calculateTeraStab: non-Tera standard STAB (1.0x, 1.5x, 2.0x w/ Adaptability)
- [x] calculateTeraStab: Tera match+original = 2.0x, Tera-only or original-only = 1.5x
- [x] calculateTeraStab: Adaptability 2.0x->2.25x (Tera+original), 1.5x->2.0x (Tera-only), NO boost for original-only
- [x] calculateTeraStab: Stellar one-time 2.0x per base type, 1.5x after consumed, 4915/4096 non-base
- [x] Tera state persists through switches via PokemonInstance fields
- [x] All 100 gen9 tests pass, typecheck clean, biome clean

Closes: N/A

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Gen 9 Terastallization added: usable in battle, transforms Tera Blast, and persists terastallized state across switches
  * STAB/damage logic updated to handle Tera, Stellar one-time boosts, and Adaptability interactions

* **Tests**
  * Added end-to-end and smoke tests covering availability, activation, move transformation, persistence, and STAB behavior

* **Documentation**
  * Updated Gen 9 progress/status to include Wave 2 Terastallization and test counts

### CODE REVIEW

File: specs/reference/gen9-status.md [hunk 1]
```diff
@@ -28,15 +28,17 @@
...
+| 0 | #715 | Package scaffold, data files, type chart, crit calc, smoke tests |
 
 ---
 
 ## IN PROGRESS
 
 | Item | Branch | Notes |
 |------|--------|-------|
-| Wave 0 scaffold | feat/gen9-wave0-scaffold | Package structure, data, battle pkg changes |
+| Wave 2 Terastallization | feat/gen9-wave2-terastallization | Gen9Terastallization gimmick + calculateTeraStab + 44 tests |
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply specs/reference/gen9-status.md]: Fixed -- added PR #718 to the Wave 2 row in gen9-status.md in commit 80d8024.
